presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -1457,6 +1462,76 @@ protected Node aggregateResult(Node aggregate, Node nextResult)
         throw new UnsupportedOperationException("not yet implemented");
     }
 
+    private static String decodeString(SqlBaseParser.UnicodeStringContext context)
```

Comment:
I'm not convinced about this decoding happening in the parser. First, it means that unicode strings don't roundtrip. If someone creates a view with one of these strings, SHOW CREATE VIEW will not recover the string they originally input (or even anything resembling that). With binary literals we try to, at least, produce a proper X'....' string.